### DISCUSSION TITLE
Pre-Amazon account tokens

### DISCUSSION THREAD

**Discussion by audiamus** [mkb79/Audible#88] [Q&A] [answered]
I am following your advice and modify my code to register a device with the Audible URL for a pre-Amazon account.

My implementation also refreshes the access token once expired, although I have noticed that the access token is hardly ever needed. The regular API calls are all signed with the adp token, it appears. Do I really need to refresh the access token if it is not used? To signal some sort of stay-alive?

And, if I still have to keep refreshing the access token, for the pre-Amazon account, I guess, I would use the Audible URL for the refresh request, too. Is that correct?

**Comment by mkb79** [comment, marked as answer]
> Do I really need to refresh the access token if it is not used? To signal some sort of stay-alive?

You only need a valid access token for a limited number of cases (e.g. deregistration, get the Amazon/Audible user profile) if you make use of the adp token request signing. Therefore a general refresh is not needed. I do not refreshed my access token for month and my auth data are still working!

> And, if I still have to keep refreshing the access token, for the pre-Amazon account, I guess, I would use the Audible URL for the refresh request, too. Is that correct?

I does not have a pre-Amazon account, so I does not have thinking about this yet! But I checked my logs from another user from US. And you are right, the refresh request was made to `https://api.audible.com/auth/token` and the deregistration to `https://api.audible.com/auth/deregister`! Without you, I wouldn't have thought about paying attention to it! Thank you very much!

**Reply by audiamus** [reply]
Thanks a lot. Based on your experience, I will change the token refresh policy in my implementation to on demand only.

> But I checked my logs from another user from US. And you are right, ...

I do not have a pre-Amazon account either, so we have to depend on feedback from users who do.
